DataFormats/TrackReco/src/TrackBase.cc

```diff
@@ -144,3 +144,20 @@ TrackBase::TrackAlgorithm TrackBase::algoByName(const std::string &name) {
   // cast
   return TrackAlgorithm(index);
 }
+
+double TrackBase::dxyError(Point const &vtx, math::Error<3>::type const &vertexCov) const {
+  // Gradient of TrackBase::dxy(const Point &myBeamSpot) with respect to track parameters
+  TrackBase::ParameterVector trackGrad =
+      TrackBase::ParameterVector(0,
+                                 0,
+                                 (vtx.x() * px() + vtx.y() * py()) / pt(),  // x_vert * cos(phi) + y_vert * sin(phi)
+                                 1,
+                                 0);
+  // Gradient with respect to point parameters
+  math::Vector<3>::type pointGrad = math::Vector<3>::type(py() / pt(),   // sin(phi)
+                                                          -px() / pt(),  // -cos(phi)
+                                                          0);
+  // Propagate covariance assuming cross-terms of the covariance between track and vertex parameters are 0
+  return std::sqrt(ROOT::Math::Dot(trackGrad, covariance() * trackGrad) +
+                   ROOT::Math::Dot(pointGrad, vertexCov * pointGrad));
```

Comment:
both can be expanded; at least the first term should be to avoid the 5x5 matrix operations, while only 2x2 is actually touched.

Reply:
Done, added a comment to at least make it easier to understand the formula.